app/code/Magento/Email/Model/Template/Filter.php

```diff
@@ -1121,7 +1121,7 @@ public function filter($value)
             $this->resetAfterFilterCallbacks();
 
             if ($this->_appState->getMode() == State::MODE_DEVELOPER) {
-                $value = sprintf(__('Error filtering template: %s'), $e->getMessage());
+                $value = sprintf(__('Error filtering template: %s')->render(), $e->getMessage());
             } else {
                 $value = (string) __("We're sorry, an error has occurred while generating this content.");
```

Comment:
Also I recommend to change this line with `render()` usage to be the same with 1124 line